chalicelib_fourfront/checks/wfr_checks.py

```diff
@@ -147,10 +149,26 @@ def md5run_status(connection, **kwargs):
     not_switched_status_to_wait = []
     # multiple failed runs
     problems = []
-    my_s3_util = s3Utils(env=connection.ff_env)
-    raw_bucket = my_s3_util.raw_file_bucket
-    out_bucket = my_s3_util.outfile_bucket
-    for a_file in res:
+    # 2024-02-08: Per above comment WRT any() on a generator, only do
+    # this S3 access only if we have at least one result in the loop below.
+    # my_s3_util = s3Utils(env=connection.ff_env)
+    # raw_bucket = my_s3_util.raw_file_bucket
+    # out_bucket = my_s3_util.outfile_bucket
+    my_s3_util = None
+    while True:
+        try:
+            # TODO: Be nicer to have a generator wrapper to allow easily
+            # checking for any results without skipping past the first item.
+            a_file = next(res)
+            if not my_s3_util:
+                my_s3_util = s3Utils(env=connection.ff_env)
+                raw_bucket = my_s3_util.raw_file_bucket
+                out_bucket = my_s3_util.outfile_bucket
```

Comment:
This you definitely can put outside the loop no?

Reply:
I think no reason why not but if nothing is returned by the search then it's done for nothing - at least I guess that is what David was trying to avoid by checking and then creating if not there yet within the loop?